src/App.tsx

```diff
@@ -15,36 +16,98 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  BY_LENGTH = 'BY_LENGTH',
+  BY_ALPHABET = 'BY_ALPHABET',
+  DEFAULT = 'DEFAULT',
+}
+
+interface SortOptions {
+  sortField: SortType;
+  isReversed: boolean;
+}
+
+function preparedList(goods: string[], { sortField, isReversed }: SortOptions) {
+  let prepared = [...goods];
+
+  if (sortField) {
+    switch (sortField) {
+      case SortType.BY_ALPHABET:
+        prepared.sort((item1, item2) => item1.localeCompare(item2));
+        break;
+      case SortType.BY_LENGTH:
+        prepared.sort((item1, item2) => item1.length - item2.length);
+        break;
+      case SortType.DEFAULT:
+        prepared = [...goods];
+        break;
+    }
+  }
+
+  if (isReversed) {
+    prepared.reverse();
+  }
+
+  return prepared;
+}
```

Comment:
The function's name `preparedList` does not start with a verb, which is not consistent with best practices for naming functions. Consider renaming it to better reflect what the function does, such as `prepareSortedList` to make it clear that the function is preparing and sorting a list.